rc/lsp.kak

```diff
@@ -77,6 +77,8 @@ declare-option -docstring "Automatically highlight references with Reference fac
 declare-option -docstring "Show available code actions (default: a 💡 in the modeline)" bool lsp_auto_show_code_actions false
 # Set it to a positive number to limit the size of the lsp-hover output. Use 0 to disable the limit.
 declare-option -docstring "Set it to a positive number to limit the size of the lsp hover output. Use 0 to disable the limit" int lsp_hover_max_lines 20
+declare-option -docstring "Set it to a positive number to limit the information in the lsp hover output. Use 0 to disable the limit" int lsp_hover_max_info_lines 20
+declare-option -docstring "Set it to a positive number to limit the diagnostics in the lsp hover output. Use 0 to disable the limit" int lsp_hover_max_diagnostic_lines 20
```

Comment:
So the fix to `lsp_hover_max_lines` is already a good improvement worth taking.

I noticed that if we have a ton of diagnostics, this will show 20 lines of diagnostics and no line of hover info.
I think it would be better to show a bit of both, maybe 10 lines each.

I'd expect the behavior to be: if truncation is happening, include all
diagnostics up to a maximum of say 10 lines, and fill up what's left of the
20 line budget with hover info. After that, if there's still something left
in the budget, fill it up with the remaining diagnostics.

We can almost get there with your new options by setting
`lsp_hover_max_diagnostic_lines` to 10; however, if there is no hover info
then we do want to show up to 20 lines worth of diagnostics.

I don't think this edge case is very important.  But I do think making it
work should be the goal because that would mean we might get away without
the two new config options (or at least, most users wouldn't ever need to
use it to fix this case). As an intermediate solution we could default
`lsp_hover_max_diagnostic_lines` to something like 10?